src/scala/ly/stealth/mesos/kafka/Scheduler.scala

```diff
@@ -87,8 +87,14 @@ object Scheduler extends org.apache.mesos.Scheduler {
         "host.name" -> offer.getHostname
       )
 
-      if (HttpServer.kafkaVersion.compareTo(new Version("0.9")) >= 0)
-        defaults += ("listeners" -> s"PLAINTEXT://:${reservation.port}")
+      if (HttpServer.kafkaVersion.compareTo(new Version("0.9")) >= 0) {
+        if (broker.bindAddress != null) {
+          defaults += ("listeners" -> s"PLAINTEXT://${offer.getHostname}:${reservation.port}")
```

Comment:
@steveniemitz 
Thanks for your comments. I misunderstood something in your first post.
I discarded changes for scheduler.scala and added logic in Broker.options() to set "listeners" with result of bindAddress.resolve(). I tested the changes in my environment.
Any other comments please let me know.